Summers_James/detail.py

```diff
@@ -0,0 +1,47 @@
+from django.http import HttpResponseRedirect
+from django_mako_plus import view_function, jscontext
+from django import forms
+import re
+from catalog import models as cmod
+import math
+from django.conf import settings
+from datetime import datetime, timezone
+from django.shortcuts import render
+
+
+@view_function
+def process_request(request, prod_id=None):
+    product = cmod.Product.objects.filter(id=prod_id)[0]
```

Comment:
I would put the creation of the product variable in a try except block so the website will not break if an invalid product id is put in the /detail url

Reply:
Good idea to strength the code.